# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/DebugAPI.h
# Commit: e802face8e04
# Full Hash: e802face8e042d03933c1495d36fd02a9def7dd9
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-07-25 21:51:57
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1554524 Part 4 - Add Debugger interface for instrumenting scripts, r=jimb,tcampbell.
# ==============================================================================

diff -r fc204daba6f3 -r e802face8e04 js/src/debugger/DebugAPI.h
--- a/js/src/debugger/DebugAPI.h	Mon Jul 08 09:39:52 2019 -1000
+++ b/js/src/debugger/DebugAPI.h	Mon Jul 08 09:41:21 2019 -1000
@@ -284,6 +284,15 @@
   // Whether any debugger is observing debugger statements in a realm.
   static bool hasDebuggerStatementHook(GlobalObject* global);
 
+  /*
+   * Get any instrumentation ID which has been associated with a script using
+   * the specified debugger object.
+   */
+  static bool getScriptInstrumentationId(JSContext* cx,
+                                         HandleObject dbgObject,
+                                         HandleScript script,
+                                         MutableHandleValue rval);
+
  private:
   static bool stepModeEnabledSlow(JSScript* script);
   static bool hasBreakpointsAtSlow(JSScript* script, jsbytecode* pc);